.github/workflows/cross-arch-build-test.yml

```diff
@@ -0,0 +1,314 @@
+name: Cross-Architecture Build Test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  # Build all architectures in one place
+  build-all:
+    name: Build All Architectures
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version-file: 'go.mod'
+        cache: true
+
+    - name: Build all architectures using make
+      run: |
+        echo "Building all architectures..."
+        make release-build
+        echo "✅ All builds completed"
+
+    - name: List generated binaries
+      run: |
+        echo "Generated binaries:"
+        ls -la kubectl-oadp-*
+        echo ""
+        echo "File details:"
+        file kubectl-oadp-* || true
+
+    - name: Upload Linux binaries
+      uses: actions/upload-artifact@v4
+      with:
+        name: linux-binaries
+        path: |
+          kubectl-oadp-linux-amd64
+          kubectl-oadp-linux-arm64
+
+    - name: Upload macOS binaries  
+      uses: actions/upload-artifact@v4
+      with:
+        name: macos-binaries
+        path: |
+          kubectl-oadp-darwin-amd64
+          kubectl-oadp-darwin-arm64
+
+    - name: Upload Windows binaries
+      uses: actions/upload-artifact@v4
+      with:
+        name: windows-binaries
+        path: |
+          kubectl-oadp-windows-amd64.exe
+          kubectl-oadp-windows-arm64.exe
+
+    - name: Run host tests
+      run: |
+        echo "Running tests on build host..."
+        make test
+        echo "✅ Host tests passed"
+
+  # Test Linux binaries on native Linux runners
+  test-linux:
+    name: Test Linux
+    needs: build-all
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            arch: amd64
+            binary: kubectl-oadp-linux-amd64
+          - os: ubuntu-24.04-arm
+            arch: arm64  
+            binary: kubectl-oadp-linux-arm64
+    
+    runs-on: ${{ matrix.os }}
+    
+    steps:
+    - name: Checkout code (for tests)
+      uses: actions/checkout@v4
+
+    - name: Set up Go (for running tests)
+      uses: actions/setup-go@v5
+      with:
+        go-version-file: 'go.mod'
+        cache: true
+
+    - name: Download Linux binaries
+      uses: actions/download-artifact@v4
+      with:
+        name: linux-binaries
+
+    - name: Verify architecture match
+      run: |
+        echo "Expected architecture: ${{ matrix.arch }}"
+        echo "Runner architecture: $(uname -m)"
+        echo "Go architecture: $(go env GOARCH)"
+
+    - name: Test binary execution
```

Comment:
Same error handling concern as above. Consider making the error handling explicit with `&&` chaining or an if statement.